src/core/basetypes/MCData.cpp

```diff
@@ -42,6 +43,18 @@ void Data::allocate(unsigned int length, bool force)
     if (length <= mAllocated)
         return;
 
+    if (mBytesDeallocator) {
+        if (mBytes) {
+            unsigned int bytes_len = mLength;
+            char * bytes = strndup(mBytes, mLength);
```

Comment:
Please don't use strndup() since it won't behave correctly if there's a \0 in the data.
use malloc() and memcpy().